packages/template-retail-react-app/app/components/field/index.jsx

```diff
@@ -25,131 +25,142 @@ import {
 } from '@salesforce/retail-react-app/app/components/icons'
 import {useIntl} from 'react-intl'
 
-const Field = ({
-    name,
-    label,
-    formLabel,
-    type = 'text',
-    options = [],
-    rules = {},
-    error,
-    placeholder,
-    inputProps,
-    control,
-    autoComplete,
-    defaultValue,
-    helpText,
-    children
-}) => {
-    const intl = useIntl()
-    const [hidePassword, setHidePassword] = useState(true)
-    const PasswordIcon = hidePassword ? VisibilityIcon : VisibilityOffIcon
-    const passwordIconLabel = hidePassword
-        ? intl.formatMessage({
-              id: 'field.password.assistive_msg.show_password',
-              defaultMessage: 'Show password'
-          })
-        : intl.formatMessage({
-              id: 'field.password.assistive_msg.hide_password',
-              defaultMessage: 'Hide password'
-          })
-    const inputType =
-        type === 'password' && hidePassword ? 'password' : type === 'password' ? 'text' : type
-    return (
-        <Controller
-            name={name}
-            control={control}
-            rules={rules}
-            defaultValue={defaultValue}
-            render={({field: {onChange, value, ref}}) => {
-                const _inputProps =
-                    typeof inputProps === 'function' ? inputProps({value, onChange}) : inputProps
+const Field = forwardRef(
+    (
+        {
+            name,
+            label,
+            formLabel,
+            type = 'text',
+            options = [],
+            rules = {},
+            error,
+            placeholder,
+            inputProps,
+            control,
+            autoComplete,
+            defaultValue,
+            helpText,
+            children
+        },
+        ref
+    ) => {
+        const intl = useIntl()
+        const [hidePassword, setHidePassword] = useState(true)
+        const PasswordIcon = hidePassword ? VisibilityIcon : VisibilityOffIcon
+        const passwordIconLabel = hidePassword
+            ? intl.formatMessage({
+                  id: 'field.password.assistive_msg.show_password',
+                  defaultMessage: 'Show password'
+              })
+            : intl.formatMessage({
+                  id: 'field.password.assistive_msg.hide_password',
+                  defaultMessage: 'Hide password'
+              })
+        const inputType =
+            type === 'password' && hidePassword ? 'password' : type === 'password' ? 'text' : type
 
-                return (
-                    <FormControl id={name} isInvalid={error}>
-                        {!['checkbox', 'radio', 'hidden'].includes(type) &&
-                            (formLabel || <FormLabel>{label}</FormLabel>)}
+        return (
+            <Controller
+                name={name}
+                control={control}
+                rules={rules}
+                defaultValue={defaultValue}
+                render={({field: {onChange, value}}) => {
```

Comment:
Everything looks good to me, just wanted to double check the changes in this file though since it's hard to tell exactly what's changed with the change in indentation:

1. Field component has been updated to use `forwardRef`
2. this render prop from the `<Controller>` component no longer passes in ref but instead uses the ref from `forwardRef`

Are there any other changes that I'm missing?

Reply:
I pivoted to use the prop inputRef instead of forwardRef